common/src/main/java/io/pravega/common/concurrent/SequentialAsyncProcessor.java

```diff
@@ -70,6 +81,19 @@ public void runAsync() {
             while (canContinue) {
                 try {
                     this.runnable.run();
+                    this.consecutiveFailedAttempts.set(0);
+                } catch (Throwable ex) {
+                    int c = this.consecutiveFailedAttempts.incrementAndGet();
+                    if (ExceptionHelpers.mustRethrow(ex)) {
+                        close();
+                    }
+
+                    boolean retry = this.errorHandler.apply(ex, c);
+                    if (retry) {
+                        synchronized (this) {
+                            this.runAgain = true;
```

Comment:
the purpose of `runAgain` is to register a request that someone else wants to invoke this task while it's already running (the processor, by definition, won't run the task multiple times concurrently). If such a request comes along, it will be recorded in this variable and the `while` loop will make sure to reinvoked the task exactly once.

In this `catch` block, I am making use of it since the error handler asked us to re-run the task one more time.